synapse/storage/database.py

```diff
@@ -326,6 +324,16 @@ def _make_sql_one_line(self, sql: str) -> str:
         "Strip newlines out of SQL so that the loggers in the DB are on one line"
         return " ".join(line.strip() for line in sql.splitlines() if line.strip())
 
+    # TODO(ParamSpec):
+    #     Once Mypy supports Concatenate, this could be written as
+    #     def _do_execute(
+    #         self,
+    #         func: Callable[Concatenate[str, P], R],
+    #         sql: str,
+    #         *args: P.args,
+    #         **kwargs: P.kwargs,
+    #     ) -> R:
+    #     so long as we also pass kwargs.
```

Comment:
Seems better as an issue that we file?

Reply:
yeah, if you think so, could do that instead.